CONTRIBUTING.md

```diff
@@ -24,6 +24,10 @@ Starting with version 4.0.0, `typing_extensions` uses
 [Semantic Versioning](https://semver.org/). See the documentation
 for more detail.
 
+## Development versions
+After meaningful updates, but without concrete plans for a release, the version can pre increased once in [pyproject.toml](/pyproject.toml) file and appended with a `.dev` suffix, e.g. `4.0.1.dev`.
+Further development versions adjust their suffix in following increasing way `.dev1`, `.dev2`, ...
```

Comment:
I would suggest that we (and by that I mean our release manager, @JelleZijlstra), increase the version to `.dev` after each release, even without any updates in between. This makes is clear that the current version is not identical to the released PyPI package, and we don't forget to increment the version number after a "meaningful" update.